packages/go/cypher/models/pgsql/pgtypes.go

```diff
@@ -294,30 +294,24 @@ func ValueToDataType(value any) (DataType, error) {
 	case time.Duration:
 		return Interval, nil
 
-	case int8, int16:
+	case uint8, int8, int16:
 		return Int2, nil
 
-	case []int8, []int16:
+	case []uint8, []int8, []int16:
 		return Int2Array, nil
 
-	case int32, graph.ID:
+	case uint16, int32, graph.ID:
 		return Int4, nil
 
-	case []int32, []graph.ID:
+	case []uint16, []int32, []graph.ID:
 		return Int4Array, nil
 
-	case int, int64:
+	case uint32, uint, uint64, int, int64:
 		return Int8, nil
 
-	case []int, []int64:
+	case []uint32, []uint, []uint64, []int, []int64:
```

Comment:
As mentioned in live code review, please document your reasoning here around overflow

Reply:
Noted. I'd like to come back to this in another changeset.